### ISSUE TITLE
specGrpRef for new elements

### ISSUE THREAD

**Issue by jamescummings** [TEIC/Stylesheets#175]
When I embed an elementSpec in my ODD prose I reference it from inside my schemaSpec with 
`<specGrpRef target="#damage-spec"/>` to point to the xml:id on my elementSpec in the prose.  However when I create a brand new element in a different namespace such as:
`<elementSpec ident="specialName" mode="add" prefix="jc_" xml:lang="en" ns="http://james.blushingbunny.net/ns.html" xml:id="specialName-spec"> 
...` then I can't use specGrpRef to reference it. To not be able to do so diminishes the ability of ODD since it is especially new elements like this that we should be documenting in the prose.  

However, I may have overlooked something (like I need to prefix the ID with something when it is a new element?) or something. I've not investigated the XSLT yet.

**Comment by lb42** [comment]
A specgrpref can only reference a specgrp as the name suggests. Is your new element spec inside a specgrp?

**Comment by jamescummings** [comment]
That's not true ... I think ... I don't wrap any of my existing elementSpec in a specGrp.   Never have.  But I'll try that, thanks!

James

--
Dr James Cummings, Academic IT Services, University of Oxford

On 1 Oct 2016 11:02, Lou <notifications@github.com> wrote:

A specgrpref can only reference a specgrp as the name suggests. Is your new element spec inside a specgrp?

**Comment by lb42** [comment]
If pointing directly at a *spec rather than a specGrp works at the moment , then that is a bug since the doc says otherwise

**Comment by jamescummings** [comment]
Fair enough. I should check out the xslt to see what is happening.

--
Dr James Cummings, Academic IT Services, University of Oxford

On 1 Oct 2016 11:10, Lou <notifications@github.com> wrote:

If pointing directly at a *spec rather than a specGrp works at the moment , then that is a bug since the doc says otherwise

**Comment by jamescummings** [comment]
Embedding the new elementSpec in a specGrp doesn't seem to work. I've tried with the id on the elementSpec inside a specGrp and with the id on the specGrp itself.   Neither seem to give me the new element, whereas embedding the new elementSpec in schemaSpec works as expected. 

Pointing directly to a *Spec does work (if it is an existing element or whatever being changed). But I can't seem to get it to work inside or outside a specGrp  if it is a new element. That is only working for me when I embed the new elementSpec directly in schemaSpec.

Maybe I'm doing something else wrong, I'll try to test at some point and post a sample ODD.
